File: workout-tracker/src/app/workouts/models/exercise.ts
import { IPublicEntity } from 'app/shared/interfaces/i-public-entity';
import { NamedEntity } from '../../shared/models/named-entity';
import { ExerciseTargetAreaLink } from './exercise-target-area-link';

export class Exercise extends NamedEntity implements IPublicEntity {
  publicId: string | null = null; //GUID
  description: string = '';
  exerciseTargetAreaLinks: ExerciseTargetAreaLink[] = [];
  setup: string = '';
  movement: string = '';
  pointsToRemember: string = '';
  resistanceType: number = 0;
  oneSided: boolean = false;
  bandsEndToEnd: boolean | null = null;
  involvesReps: boolean = false;
  usesBilateralResistance: boolean = false;
}

File: API/WorkoutTracker.Domain/Exercises/Exercise.cs
using System;
using System.Collections.Generic;
using WorkoutTracker.Domain.BaseClasses;
using WorkoutTracker.Domain.Interfaces;

namespace WorkoutTracker.Domain.Exercises
{
    public class Exercise : NamedEntity, IPublicEntity
    {
        public Guid PublicId { get; set; }
        public string Description { get; set; }
        public string Setup { get; set; }
        public string Movement { get; set; }
        public string PointsToRemember { get; set; }

        public virtual ICollection<ExerciseTargetAreaLink> ExerciseTargetAreaLinks { get; set; }

        public ResistanceType ResistanceType { get; set; }
        public bool OneSided { get; set; }
        public bool? BandsEndToEnd { get; set; }
        public bool InvolvesReps { get; set; }
        public bool UsesBilateralResistance { get; set; }
    }
}
